core/src/components/refresher/test/basic/e2e.ts

```diff
@@ -1,10 +1,53 @@
+import type { E2EPage } from '@stencil/core/testing';
 import { newE2EPage } from '@stencil/core/testing';
 
-test('refresher: basic', async () => {
-  const page = await newE2EPage({
-    url: '/src/components/refresher/test/basic?ionic:_testing=true'
+import { pullToRefresh } from '../test.utils';
+
+describe('refresher: basic', () => {
+
+  let page: E2EPage;
+
+  beforeEach(async () => {
+    page = await newE2EPage({
+      url: '/src/components/refresher/test/basic?ionic:_testing=true'
+    });
+  });
+
+  it('should match existing visual screenshots', async () => {
+    const compare = await page.compareScreenshot();
+    expect(compare).toMatchScreenshot();
+  });
+
+  it('should load more items when performing a pull-to-refresh', async () => {
+    const initialItems = await page.findAll('ion-item');
+    expect(initialItems.length).toBe(30);
+
+    await pullToRefresh(page);
+
+    const items = await page.findAll('ion-item');
+    expect(items.length).toBe(60);
+  });
+
+  describe('legacy refresher', () => {
+
+    it('should load more items when performing a pull-to-refresh', async () => {
+      const refresherContent = await page.$('ion-refresher-content');
+      refresherContent.evaluate((el: any) => {
+        // Resets the pullingIcon to enable the legacy refresher
```

Comment:
I think you mean native refresher here? When `pullingIcon=undefined` we use the native refresher where possible. When pullingIcon is defined we fallback to legacy.

Reply:
Good catch! In the future, I'd like to consider our `basic` test examples as what the drop-in example looks like. So in this case, we would have native refreshers be the base example and then write a test to verify modifications/alt-states.